### PR TITLE
ENH: Use VSCode workspaces to open projects

### PR DESCRIPTION
**Pull request by ant0nsc**
* Enable VSCode workspaces for all 3 projects
* Enable opening projects individually
* Separated tool configs (flake, mypy) per project
* Removed Pycharm configs

### CODE REVIEW

File: hi-ml-azure/testazure/testazure/test_azure_util.py [hunk 1]
```diff
@@ -737,13 +737,13 @@ def test_pip_include_1() -> None:
     file in the repository.
     """
     if paths.is_himl_used_from_git_repo():
-        root_yaml = paths.git_repo_root_folder() / paths.ENVIRONMENT_YAML_FILE_NAME
-        assert root_yaml.is_file()
-        original_yaml = conda_merge.read_file(root_yaml)
+        yaml = paths.shared_himl_conda_env_file()
```
Comment (fepegar) [inline hi-ml-azure/testazure/testazure/test_azure_util.py:740]: I suggest a more informative variable name, e.g., `environment_path`.

**Review by fepegar — APPROVED** [review]

File: docs/source/developers.md [hunk 2]
```diff
@@ -7,23 +7,39 @@ On Windows system, we recommend using WSL, the Windows Subsystem for Linux, beca
...
+* The `himl` Conda environment should be used for work on the `hi-ml` and `hi-ml-azure` projects.
+* The `HimlHisto` Conda environment should be used for work on `hi-ml-histopathology`.
+
+Please select the right Python interpreter for your project (or all projects if using the `himl-projects` workspace)
+inside VSCode, by choosing "Python: Select Interpreter" from the command palette (Ctrl-Shift-P on VSCode for Windows)
+
+To create the Conda environment `himl`, please use either
 
 ```shell script
 conda env create --file environment.yml
```
Comment (mebristo) [inline docs/source/developers.md:33]: Should this be updated to `hi-ml/environment.yml`?

File: hi-ml-azure/.vscode/launch.json [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            "module": "testazure.test_data.make_tests",
+            "cwd": "${workspaceFolder}/testazure/",
+            "console": "integratedTerminal"
+        },
+        {
+            "name": "Example Hyperdrive run in AzureML",
+            "type": "python",
+            "request": "launch",
+            "program": "${workspaceFolder}/docs/source/examples/8/sample.py",
+            "cwd": "${workspaceFolder}/docs/source/examples/8/",
```
Comment (mebristo) [inline hi-ml-azure/.vscode/launch.json:56]: This suggests there is a folder `hi-ml/hi-ml-azure/docs`, but as far as I can see the docs still live at `hi-ml/docs`?

File: hi-ml-azure/testazure/testazure/test_azure_util.py (hunk 1 above)
Comment (mebristo) [inline hi-ml-azure/testazure/testazure/test_azure_util.py:740]: Could rename this so as not to confuse with pyyaml's `yaml` module?

File: hi-ml-histopathology/.mypy.ini [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+scripts_are_modules=True
+namespace_packages=True
+show_traceback=True
+ignore_missing_imports=True
+follow_imports=normal
+follow_imports_for_stubs=True
+disallow_untyped_calls=False
+disallow_untyped_defs=True
+strict_optional=True
+mypy_path=$MYPY_CONFIG_FILE_DIR/src:$MYPY_CONFIG_FILE_DIR/../hi-ml/src:$MYPY_CONFIG_FILE_DIR/../hi-ml-azure/src:
```
Comment (mebristo) [inline hi-ml-histopathology/.mypy.ini:12]: Does this mean that running mypy in hi-ml-histopathology will also run mypy on hi-ml and hi-ml-azure?

File: hi-ml-histopathology/.vscode/settings.json [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        "--src=${workspaceFolder}/../hi-ml/src",
+        "--src=${workspaceFolder}/../hi-ml-azure/src",
+        "--src=${workspaceFolder}/src"
+    ],
+    "python.testing.pytestArgs": [
+        "${workspaceFolder}/testhisto",
+        "${workspaceFolder}/testSSL"
+    ],
+    "python.analysis.extraPaths": [
+        "./hi-ml-azure/src",
```
Comment (mebristo) [inline hi-ml-histopathology/.vscode/settings.json:62]: Are these unaffected by the workspaceFolder?

File: docs/source/developers.md (hunk 2 above)
Comment (ant0nsc) [reply docs/source/developers.md:33]: yes, thanks!

File: hi-ml-azure/.vscode/launch.json (hunk 3 above)
Comment (ant0nsc) [reply hi-ml-azure/.vscode/launch.json:56]: correct, thanks

File: hi-ml-azure/.vscode/settings.json [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    // and may override settings in the workspace configuration.
+    "python.sortImports.args": [
+        "-l=120",
+        "--src=${workspaceFolder}/src"
+    ],
+    "python.testing.pytestArgs": [
+        "${workspaceFolder}/testazure/testazure"
+    ],
+    "python.analysis.extraPaths": [
+        "./hi-ml-azure/src",
```
[the remark this answers is not in the document]
Comment (ant0nsc) [reply hi-ml-azure/.vscode/settings.json:59]: yes!

File: hi-ml-histopathology/.mypy.ini (hunk 4 above)
Comment (ant0nsc) [reply hi-ml-histopathology/.mypy.ini:12]: no, these are the paths where mypy looks for source. And here we want to tell it about the existing of the other package code.

File: hi-ml-histopathology/.vscode/settings.json (hunk 5 above)
Comment (ant0nsc) [reply hi-ml-histopathology/.vscode/settings.json]: no, you spotted a problem, fixed!

**Pull request merged**